app.js

```diff
@@ -133,44 +124,29 @@ function renderProduct(id, title, imageUrl, price, description) {
     <h5>price: ${price}</h5>
     <p>Product id : ${id}</p>
     <p>${description}</p>
-    <button class="edit">Edit</button>
-    <button class="delete">Delete</button>
+    <button class="view" id='${id}'>View</button>
+    <button class="edit" id='${id}'>Edit</button>
+    <button class="delete" id='${id}'>Delete</button>
   </div>`;
   productList.appendChild(productElement);
   updateUI();
 }
 
-addBtn.addEventListener("click", addMovieHandler);
-
-
-// window.addEventListner("load",function(){
-//     products.forEach(product => {
-//         const productElement = document.createElement('li');
-//     productElement.classList.add('product-element');
-
-//     productElement.innerHTML = `<div class="product-image">
-//     <img src="${product.imageUrl}" alt="${product.title}">
-//   </div>
-//   <div class="product-description">
-//     <h2>${product.title}</h2>
-//     <h5>price: ${product.price}</h5>
-//     <p>Product id : ${product.price}</p>
-//     <p>${product.description}</p>
-//     <button class="edit">Edit</button>
-//     <button class="delete">Delete</button>
-//   </div>`;
-//   productList.appendChild(productElement);
-//     })
-// });
-
-console.log(products[0].price);
-
-products.forEach((product) => {
-  console.log(product.price);
-});
+addBtn.addEventListener("click", addProductHandler);
+
+window.onload = LoadProducts;
+
+function LoadProducts() {
+  const str = localStorage.getItem("ProductArray");
+
+  const parsedProducts = JSON.parse(str);
+  if(parsedProducts === null) return;
```

Comment:
I think you should check the length of the array instead of checking whether it is null or not.

Reply:
sure i'll keep in mind that from the next time.